### PR TITLE
Upgrade checkstyle and checkstyle plugin

### PR DESCRIPTION
**Pull request by chtyim**
```
- The upgrade allows ignoring the missing suppression.xml file
  - The file is copied and provided by the plugin
  - It is needed so that mvn can run from a different directory
- Also fixes the RedundantModifier problem detected by newer checkstyle
```

### CODE REVIEW

File: cdap-api/src/main/java/co/cask/cdap/api/common/Scope.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
   /**
    * Private constructor to force using the enum values.
```
Comment (wolf31o2) [inline cdap-api/src/main/java/co/cask/cdap/api/common/Scope.java:33]: This isn't private, anymore.

**Comment by chtyim** [comment]
checkstyle build passed: http://builds.cask.co/browse/CDAP-DRC4190

File: cdap-api/src/main/java/co/cask/cdap/api/common/Scope.java (hunk 1 above)
Comment (chtyim) [inline cdap-api/src/main/java/co/cask/cdap/api/common/Scope.java:33]: It is private. All enum constructor are private, that's why no need to declare it as "private" (checkstyle says redundant modifier).

**Comment by wolf31o2** [comment]
👍 LGTM

**Pull request merged**